python/EC2_SECURITY_GROUP_NOT_USED/EC2_SECURITY_GROUP_NOT_USED.py

```diff
@@ -0,0 +1,366 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+
+Description:
+  Checks that Security Groups are attached to EC2 instances or elastic network interfaces (ENIs). The rule returns NON_COMPLIANT for a Security Group if it's not associated with any Elastic Network Interface.
+Trigger:
+  Configuration Changes
+Reports on:
+  AWS::EC2::SecurityGroup
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+     Given: 'relationships' in the configuration item does not contain a Network interface Id in resourceId
+     Then: Return NON_COMPLIANT with annnotation 'This Security Group is not associated with any resource'.
+  Scenario: 2
+     Given: 'relationships' in the configuration item contains at least one Network interface Id in resourceId
+     Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EC2::SecurityGroup'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    for relation in configuration_item["relationships"]:
+        #resourceId for eni: 'eni-07cd2307b857000e6'
+        if relation['resourceId'][0:3] == "eni":
+            return build_evaluation_from_config_item(configuration_item, "COMPLIANT")
+    return build_evaluation_from_config_item(configuration_item, "NON_COMPLIANT", "The security group " + configuration_item["configuration"]["groupName"] + " is not associated with any ENI/EC2")

```

Comment:
Change annotation to use full service/resource name: "This Amazon EC2 Security Group is not associated with any Amazon Elastic Network Interface."
Also, we don't need the resource name as this is a config change rule triggered per resource, hence we already have the resourceId.